packages/flutter/test/material/dialog_theme_test.dart

```diff
@@ -548,4 +550,54 @@ void main() {
       ),
     );
   });
+
+  testWidgets('DialogTheme.clipBehavior updates the dialogs clip behavior', (WidgetTester tester) async {
+    const DialogTheme dialogTheme = DialogTheme(
+      clipBehavior: Clip.hardEdge
+    );
+    const Dialog dialog = Dialog(child: Placeholder());
+
+    await tester.pumpWidget(_appWithDialog(
+      tester,
+      dialog,
+      theme: ThemeData(dialogTheme: dialogTheme),
+    ));
```

Comment:
```suggestion
    await tester.pumpWidget(_appWithDialog(
      tester,
      dialog,
      theme: ThemeData(dialogTheme: dialogTheme),
    ));
    
    await tester.tap(find.text('X'));
    await tester.pumpAndSettle();    
```

Tap to open Dialog so the Material can be created.